tornado/websocket.py

```diff
@@ -474,33 +484,43 @@ def _abort(self):
 
 
 class _PerMessageDeflateCompressor(object):
-    def __init__(self, persistent, max_wbits):
+    def __init__(self, persistent, max_wbits, compression_options=None):
         if max_wbits is None:
             max_wbits = zlib.MAX_WBITS
         # There is no symbolic constant for the minimum wbits value.
         if not (8 <= max_wbits <= zlib.MAX_WBITS):
             raise ValueError("Invalid max_wbits value %r; allowed range 8-%d",
                              max_wbits, zlib.MAX_WBITS)
         self._max_wbits = max_wbits
+
+        if compression_options is None or not 'compression_level' in compression_options:
+            self._compression_level = tornado.web.GZipContentEncoding.GZIP_LEVEL
+        else:
+            self._compression_level = compression_options['compression_level']
+
+        if compression_options is None or not 'mem_level' in compression_options:
+            self._mem_level = 8
+        else:
+            self._mem_level = compression_options['mem_level']
+
         if persistent:
-            self._compressor = self._create_compressor()
+            self._compressor = self._create_compressor(self._compression_level, self._mem_level)
         else:
             self._compressor = None
 
-    def _create_compressor(self):
-        return zlib.compressobj(tornado.web.GZipContentEncoding.GZIP_LEVEL,
-                                zlib.DEFLATED, -self._max_wbits)
+    def _create_compressor(self, compression_level, mem_level):
```

Comment:
The arguments are `self._compression_level` and `self._mem_level` at every call site. Why not just use the member variables like we do with `self._max_wbits` instead of passing them as arguments?